### PR TITLE
replace http.request with the request library to be able to use proxies

### PR DESCRIPTION
**Pull request by HughePaul**
Added request module to enable proxies specified by environment variables:
HTTP_PROXY and HTTPS_PROXY

### CODE REVIEW

**Comment by gavboulton** [comment]
Looking.

**Comment by gavboulton** [comment]
LGTM.

File: lib/model.js [hunk 1]
```diff
@@ -110,58 +121,40 @@ _.extend(Model.prototype, {
...
+                this.emit('success', data, originalSettings, statusCode, responseTime);
             }
             if (typeof callback === 'function') {
                 callback(err, data, responseTime);
             }
         }.bind(this);
 
-        var request = protocol.request(settings, function (response) {
+        Model._request(settings, function (e, response) {
+            if (e) {
```
Comment (JoeChapman) [inline lib/model.js]: `e` === `event`?

Comment (JoeChapman) [inline lib/model.js]: BTW, I know it's `error`. Devils advocate

Comment (HughePaul) [inline lib/model.js]: fair comment, updated to `err`, across the module

**Comment by gavboulton** [comment]
hmpo-model@0.7.0 published.

**Pull request merged**